From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Sun, 3 Jul 2022 18:42:40 +0700
Subject: [PATCH] ver command stuff


diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index 6bb7607dde6c3f20f2f2b9b8b5dad877f84f6646..f509909c127ebb92490f990fa5242bce2b9d792e 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -49,7 +49,14 @@ public class VersionCommand extends BukkitCommand {
         if (!testPermission(sender)) return true;
 
         if (args.length == 0) {
-            sender.sendMessage("This server is running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " (Implementing API version " + Bukkit.getBukkitVersion() + ")");
+            //sender.sendMessage("This server is running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " (Implementing API version " + Bukkit.getBukkitVersion() + ")");
+            sender.sendMessage(translate(
+                    "&8/&7&m------------------------------------------------&8\\",
+                    "&fThis &eserver&f is running &bNeonPaper",
+                    "&fVersion&7: &b" + Bukkit.getVersion(),
+                    "&fAPI&7: &e" + Bukkit.getBukkitVersion(),
+                    "&8\\&7&m-----------------------------------------------&8/"
+            ));
             tellHistory(sender); // Paper
             sendVersion(sender); // Paper - We'll say when, thanks
         } else {
@@ -209,8 +216,8 @@ public class VersionCommand extends BukkitCommand {
     private void obtainVersion() {
         String version = Bukkit.getVersion();
         if (version == null) version = "Custom";
-        if (version.startsWith("git-Paper-")) {
-            String[] parts = version.substring("git-Paper-".length()).split("[-\\s]");
+        if (version.startsWith("git-NeonPaper-")) { // NeonPaper
+            String[] parts = version.substring("git-NeonPaper-".length()).split("[-\\s]"); // NeonPaper
             int distance = getDistance(null, parts[0]);
             switch (distance) {
                 case -1:
@@ -261,13 +268,13 @@ public class VersionCommand extends BukkitCommand {
 
     // Paper start
     private static int getDistance(String repo, String verInfo) {
-        try {
+        //try {
             int currentVer = Integer.decode(verInfo);
-            return getFromJenkins(currentVer);
-        } catch (NumberFormatException ex) {
+            /*return getFromJenkins(currentVer);
+        } catch (NumberFormatException ex) {/*/
             verInfo = verInfo.replace("\"", "");
-            return getFromRepo("PaperMC/Paper", verInfo);
-        }
+            return getFromRepo("hungnguyennaz/NeonPaper", verInfo);
+        //}
             /*
             BufferedReader reader = Resources.asCharSource(
                     new URL("https://hub.spigotmc.org/stash/rest/api/1.0/projects/SPIGOT/repos/" + repo + "/commits?since=" + URLEncoder.encode(hash, "UTF-8") + "&withCounts=true"),
@@ -285,7 +292,7 @@ public class VersionCommand extends BukkitCommand {
             */
     }
 
-    private static int getFromJenkins(int currentVer) {
+    /*private static int getFromJenkins(int currentVer) {
         try {
             BufferedReader reader = Resources.asCharSource(
                     new URL("https://ci.destroystokyo.com/job/Paper/lastSuccessfulBuild/buildNumber"), // Paper
@@ -304,10 +311,10 @@ public class VersionCommand extends BukkitCommand {
             e.printStackTrace();
             return -1;
         }
-    }
+    }*/
 
     // Contributed by Techcable <Techcable@outlook.com> in GH PR #65
-    private static final String BRANCH = "ver/1.12.2";
+    private static final String BRANCH = "main";
     private static int getFromRepo(String repo, String hash) {
         try {
             HttpURLConnection connection = (HttpURLConnection) new URL("https://api.github.com/repos/" + repo + "/compare/" + BRANCH + "..." + hash).openConnection();
