From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hungnguyennaz <no-reply@github.com>
Date: Wed, 6 Jul 2022 00:38:20 +0700
Subject: [PATCH] Reaper Fix MC-120780


diff --git a/src/main/java/net/minecraft/server/PlayerChunk.java b/src/main/java/net/minecraft/server/PlayerChunk.java
index b40c136fae6e3e5b8bd706cfaeea2e5aedad2b05..963784cbbc512761f67b685f26e0e5fab48abd27 100644
--- a/src/main/java/net/minecraft/server/PlayerChunk.java
+++ b/src/main/java/net/minecraft/server/PlayerChunk.java
@@ -141,6 +141,7 @@ public class PlayerChunk {
             this.dirtyCount = 0;
             this.h = 0;
             this.done = true;
+            if (this.c.isEmpty()) return true; // Reaper - Fix MC-120780
             PacketPlayOutMapChunk packetplayoutmapchunk = new PacketPlayOutMapChunk(this.chunk, '\uffff');
             Iterator iterator = this.c.iterator();
 
