From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hungnguyennaz <no-reply@github.com>
Date: Wed, 6 Jul 2022 00:51:29 +0700
Subject: [PATCH] Remove spigot annoying offline mode alerts


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index bf50ce29109755e8af34f0592e56f0d82a9527af..15760d19346d27bdca27096edb272cbde6c55713 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -233,6 +233,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
 
             if (!this.getOnlineMode()) {
                 DedicatedServer.LOGGER.warn("**** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!");
+                /* // NeonPaper start - Shut the fuck up spigot
                 DedicatedServer.LOGGER.warn("The server will make no attempt to authenticate usernames. Beware.");
                 // Spigot start
                 if (org.spigotmc.SpigotConfig.bungee) {
@@ -243,6 +244,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
                 }
                 // Spigot end
                 DedicatedServer.LOGGER.warn("To change this, set \"online-mode\" to \"true\" in the server.properties file.");
+                */ // NeonPaper end 
             }
 
             if (this.aS()) {
