From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Softik Lord <dimap9986@gmail.com>
Date: Sat, 9 Jul 2022 03:07:18 +0500
Subject: [PATCH] Use-Jafama


diff --git a/src/main/java/net/minecraft/server/BiomeMesa.java b/src/main/java/net/minecraft/server/BiomeMesa.java
index c5472b50109c2efb46c896d097860e48e65356ca..04c044a7182891f0bde69d65f7448d379946e663 100644
--- a/src/main/java/net/minecraft/server/BiomeMesa.java
+++ b/src/main/java/net/minecraft/server/BiomeMesa.java
@@ -2,6 +2,7 @@ package net.minecraft.server;
 
 import java.util.Arrays;
 import java.util.Random;
+import net.jafama.FastMath;
 
 public class BiomeMesa extends BiomeBase {
 
@@ -89,7 +90,7 @@ public class BiomeMesa extends BiomeBase {
         IBlockData iblockdata = BiomeMesa.A;
         IBlockData iblockdata1 = this.r;
         int j1 = (int) (d0 / 3.0D + 3.0D + random.nextDouble() * 0.25D);
-        boolean flag = MathHelper.cos((float)d0 / 3.0f * 3.141592653589793f) > 0.0f;
+        boolean flag = FastMath.cos(d0 / 3.0D * 3.141592653589793D) > 0.0D; //NeonPaper 
         int k1 = -1;
         boolean flag1 = false;
         int l1 = 0;
diff --git a/src/main/java/net/minecraft/server/ChunkProviderHell.java b/src/main/java/net/minecraft/server/ChunkProviderHell.java
index 12bc10ff00eda73bcad7c6394088d84cfbc06fcf..32cc1f1b93c9e3999e317bb47954c454ddb7782d 100644
--- a/src/main/java/net/minecraft/server/ChunkProviderHell.java
+++ b/src/main/java/net/minecraft/server/ChunkProviderHell.java
@@ -3,6 +3,7 @@ package net.minecraft.server;
 import java.util.List;
 import java.util.Random;
 import javax.annotation.Nullable;
+import net.jafama.FastMath;
 
 public class ChunkProviderHell implements ChunkGenerator {
 
@@ -246,7 +247,8 @@ public class ChunkProviderHell implements ChunkGenerator {
         int l1;
 
         for (l1 = 0; l1 < i1; ++l1) {
-            adouble1[l1] = Math.cos((double) l1 * 3.141592653589793D * 6.0D / (double) i1) * 2.0D;
+            //adouble1[l1] = Math.cos((double) l1 * 3.141592653589793D * 6.0D / (double) i1) * 2.0D;
+			adouble1[l1] = FastMath.cos((double) l1 * 3.141592653589793D * 6.0D / (double) i1) * 2.0D; //NeonPaper 
             double d2 = (double) l1;
 
             if (l1 > i1 / 2) {
diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
index 6c30d30ff2d7c46aa9e3e32473efc94cb243defa..776ec462790e18be092e1b7c2b42dbf12a852e2f 100644
--- a/src/main/java/net/minecraft/server/EntityHuman.java
+++ b/src/main/java/net/minecraft/server/EntityHuman.java
@@ -8,6 +8,7 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
 import javax.annotation.Nullable;
+import net.jafama.FastMath;
 
 // CraftBukkit start
 import org.bukkit.craftbukkit.entity.CraftHumanEntity;
@@ -527,10 +528,13 @@ public abstract class EntityHuman extends EntityLiving {
                 entityitem.motY = (double) (-MathHelper.sin(this.pitch * 0.017453292F) * f + 0.1F);
                 f1 = this.random.nextFloat() * 6.2831855F;
                 f = 0.02F * this.random.nextFloat();
-                entityitem.motX += Math.cos((double) f1) * (double) f;
-                entityitem.motY += (double) ((this.random.nextFloat() - this.random.nextFloat()) * 0.1F);
-                entityitem.motZ += Math.sin((double) f1) * (double) f;
-            }
+                //entityitem.motX += Math.cos((double) f1) * (double) f;
+				entityitem.motX += FastMath.cos((double) f1) * (double) f; // NeonPaper
+                //entityitem.motY += (double) ((this.random.nextFloat() - this.random.nextFloat()) * 0.1F);
+				entityitem.motY += (double) ((this.random.nextFloat() - this.random.nextFloat()) * 0.1F); // NeonPaper
+				//entityitem.motZ += Math.sin((double) f1) * (double) f;
+				entityitem.motZ += FastMath.sin((double) f1) * (double) f; // NeonPaper
+		  }
 
             // CraftBukkit start - fire PlayerDropItemEvent
             Player player = (Player) this.getBukkitEntity();
diff --git a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
index 8d917b9e5a804151c8c5371783e8ef6f4fdf7222..0071bec9c3b1c9ddca365ef242af70b412b6d793 100644
--- a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+++ b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
@@ -5,6 +5,7 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import javax.annotation.Nullable;
+import net.jafama.FastMath;
 
 // CraftBukkit start
 import org.bukkit.Location;
@@ -487,8 +488,10 @@ public abstract class EntityMinecartAbstract extends Entity implements INamableT
         if (entity instanceof EntityLiving) {
             d6 = (double) ((EntityLiving) entity).bg;
             if (d6 > 0.0D) {
-                d7 = -MathHelper.sin(entity.yaw * 0.017453292F);
-                d8 = MathHelper.cos(entity.yaw * 0.017453292F);
+               // d7 = -MathHelper.sin(entity.yaw * 0.017453292F);
+                //d8 = MathHelper.cos(entity.yaw * 0.017453292F);
+                d7 = -FastMath.sin((double) (this.passenger.yaw * 3.1415927F / 180.0F)); //NeonPaper
+                d8 = FastMath.cos((double) (this.passenger.yaw * 3.1415927F / 180.0F)); //NeonPaper
                 d9 = this.motX * this.motX + this.motZ * this.motZ;
                 if (d9 < 0.01D) {
                     this.motX += d7 * 0.1D;
