From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Softik Lord <dimap9986@gmail.com>
Date: Sat, 9 Jul 2022 03:44:34 +0500
Subject: [PATCH] Optimize-head-rotation


diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index 56672a69dd62afe9cb054850843524eb962ed4b4..1db66e9a4b09f6a5582ffe43ffc542c6b7240ade 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -250,7 +250,14 @@ public class EntityTrackerEntry {
 
             i = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
             if (Math.abs(i - this.headYaw) >= 1) {
-                this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+                // this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+			    // NeonPaper start
+                /* This was originally introduced by CraftBukkit, though the implementation is wrong since it's broadcasting
+                the packet again in a method that is already called for each player. This would create a very serious performance issue
+                with high player and entity counts (each sendPacket call involves waking up the event loop and flushing the network stream).
+                this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));*/
+                entityplayer.playerConnection.sendPacket(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+                // NeonPaper end
                 this.headYaw = i;
             }
 
