From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Softik Lord <dimap9986@gmail.com>
Date: Sat, 9 Jul 2022 04:18:14 +0500
Subject: [PATCH] Even-more-fast-util


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 2555d76316c534a4c31dd1d17a2638c47cbaefe9..386b7fbf68d7ebe88f45843748f4aa7d58b4a58d 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -37,6 +37,7 @@ import org.bukkit.generator.ChunkGenerator;
 // Paper start
 import java.util.Set;
 import com.google.common.collect.Sets;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
 // Paper end
 
 public abstract class World implements IBlockAccess {
@@ -44,7 +45,7 @@ public abstract class World implements IBlockAccess {
     private int a = 63;
     protected boolean d;
     // Spigot start - guard entity list from removals
-    public final List<Entity> entityList = new java.util.ArrayList<Entity>()
+    public final List<Entity> entityList = new ObjectArrayList<Entity>() // NeonPaper - ArrayList -> ObjectArrayList
     {
         @Override
         public Entity remove(int index)
@@ -74,8 +75,10 @@ public abstract class World implements IBlockAccess {
     public final List<TileEntity> tileEntityListTick = Lists.newArrayList();
     private final List<TileEntity> b = Lists.newArrayList();
     private final Set<TileEntity> tileEntityListUnload = Sets.newHashSet(); // Paper
-    public final List<EntityHuman> players = Lists.newArrayList();
-    public final List<Entity> j = Lists.newArrayList();
+    //public final List<EntityHuman> players = Lists.newArrayList();
+    //public final List<Entity> j = Lists.newArrayList();
+    public final List<EntityHuman> players = new ObjectArrayList<>(); // NeonPaper - ArrayList -> ObjectArrayList
+    public final List<Entity> k = new ObjectArrayList<>(); // NeonPaper - ArrayList -> ObjectArrayList	
     protected final IntHashMap<Entity> entitiesById = new IntHashMap();
     private final long K = 16777215L;
     private int L; private int getSkylightSubtracted() { return this.L; } // Paper - OBFHELPER
