From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Softik Lord <dimap9986@gmail.com>
Date: Mon, 11 Jul 2022 01:43:29 +0500
Subject: [PATCH] Disconnect-Spam-Options


diff --git a/src/main/java/me/hungnguyennaz/neonpaper/NeonPaperConfig.java b/src/main/java/me/hungnguyennaz/neonpaper/NeonPaperConfig.java
index e1f21fa24e214551afdf8e9eb1490906986ea08b..0573122cc243a787e2363925f51d04a44b995f0e 100644
--- a/src/main/java/me/hungnguyennaz/neonpaper/NeonPaperConfig.java
+++ b/src/main/java/me/hungnguyennaz/neonpaper/NeonPaperConfig.java
@@ -71,6 +71,18 @@ public class NeonPaperConfig {
         }
     }
 
+    public static boolean disconnectSpam;
+    private static void disconnectSpam()
+    {
+        disconnectSpam = getBoolean("disconnect-spam.enabled", true);
+    }
+
+    public static int disconnectSpamAddition;
+    private static void disconnectSpamAddition()
+    {
+        disconnectSpamAddition = getInt("disconnect-spam.addition", 5);
+    }
+
 
     static void readConfig(Class<?> clazz, Object instance) {
         for (Method method : clazz.getDeclaredMethods()) {
diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 25c3ff1ce163a8887ccde81b1b9c4e1a8c6ac7ba..45e0aacfc7dca54bbb60cbe0a66ed284f93859d0 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -10,6 +10,7 @@ import io.netty.util.concurrent.GenericFutureListener;
 import java.io.IOException;
 import java.util.*;
 
+import me.hungnguyennaz.neonpaper.NeonPaperConfig;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -1357,7 +1358,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
             // Spigot end
             // CraftBukkit start - replaced with thread safe throttle
             // this.chatThrottle += 20;
-            if (counted && chatSpamField.addAndGet(this, 20) > 200 && !this.minecraftServer.getPlayerList().isOp(this.player.getProfile())) { // Spigot
+            if (NeonPaperConfig.disconnectSpam && counted && chatSpamField.addAndGet(this, NeonPaperConfig.disconnectSpamAddition) > 200 && !this.minecraftServer.getPlayerList().isOp(this.player.getProfile())) { // Spigot
                 if (!isSync) {
                     Waitable waitable = new Waitable() {
                         @Override
@@ -1376,7 +1377,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                     } catch (ExecutionException e) {
                         throw new RuntimeException(e);
                     }
-                } else {
+                } else if (NeonPaperConfig.disconnectSpam) {
                     this.disconnect(new ChatMessage("disconnect.spam", new Object[0]));
                 }
                 // CraftBukkit end
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 6750a96c275c7b4399124f600b85d30c40f86392..c45eaecf8aec8802e5b2dfbeea61b3115e9be3ca 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -169,6 +169,18 @@ public class CraftWorld implements World {
         });
     }
 
+    public static boolean disconnectSpam;
+    private static void disconnectSpam()
+    {
+        disconnectSpam = getBoolean("disconnect-spam.enabled", true);
+    }
+
+    public static int disconnectSpamAddition;
+    private static void disconnectSpamAddition()
+    {
+        disconnectSpamAddition = getInt("disconnect-spam.addition", 5);
+    }
+
     public void getChunkAtAsync(Block block, ChunkLoadCallback callback) {
         getChunkAtAsync(block.getX() >> 4, block.getZ() >> 4, callback);
     }
